Store visibility status of sidebar menus in localstorage

PolarizedIons opened the pull request:
Closes #51. The example only talked about the userlist, but keeping with the title, I did both that and the network list, split into separate commits if the latter isn't needed/wanted.

xPaw commented on client/js/lounge.js:
```diff
@@ -411,6 +422,18 @@ $(function() {
+			viewport.find(".chan.active .rt").click();
+		}
+
+		// Toggle networklist if its "openness" isn't what it should be (what is stored in localstorage)
```
You don't really need to sync network list here (just on page load should be enough?)

PolarizedIons replied:
...right. I didn't put much thought into where I placed this one, was just thinking "put this next to this one" :+1:

PolarizedIons commented on client/js/utils.js:
```diff
@@ -85,3 +86,7 @@ function requestIdleCallback(callback, timeout) {
 }
+
+function isUserlistVisible() {
+	return $(".chan.active .sidebar").first().css("right") === "0px";
```
I know this isn't the most foolproof way of doing this, and it works. Does anyone have any objections/suggestions for this?

xPaw commented:
@PolarizedIons Do you want to also look into making network hiding work on desktop? Because the button is currently hidden and you can't hide it.

PolarizedIons commented:
Alright, I first tried just doing the same as on mobile, but that moves part of the channel off of the screen, so I opted for just shrinking the channel when it's open, and expanding it when closed. The shrinking didn't look very good on mobile, so I kept that the same.

I also noticed a bug with how I was previously detecting the userlist opening (via the 'transitionend' event) when dragging: when you drag the menu open entirely (vs. just flicking it open) that event never gets triggered, so I moved that storage part into `libs/slideout.js`.

I also added a bugfix I noticed a lot while testing - the userlist opening/closing when going from "desktop" to "mobile" widths. Let me know if I should remove this.

YaManicKill commented:
So, I did a bit of investigating, and I've at least found how to get the 2 to show at the same time. If you move the network list toggle check that's on line 44-48, into init.js after openCorrectChannel, and you put the `$("#loading").remove();` straight after it, then the loading screen is remove basically at the same time as the 2 sidebars are expanded.

With your code:

![](https://files.yamanickill.com/1hRNoTr9.gif)

With this change:

![](https://files.yamanickill.com/oi_NbrS_.gif)

I've put the changes up at [this commit](https://github.com/thelounge/lounge/commit/dc6add7b8581a7f1eae03e2bf456f4c71d7dbfcc) if you want to try them out. I think this would be a good addition.

It might be nice to make them both display none, then show them if we want to, then make them display and hide the loading screen at the same time, because then you won't get the animation on the first load, and all you'll see is the loading screen disappear and the page behind appear. I think it'll be much nicer. If I have time tomorrow I might investigate that. It might need some z-index changes as well.

McInkay commented:
Hey, @PolarizedIons, would you have any time to address the comments we left, and rebase/fix conflicts?

astorije commented:
Closing this, superseded by @xPaw's #2235. Thanks for starting this, @PolarizedIons!! 🙏
